### ISSUE TITLE
Gradient loss dimension issue for 2D images in pytorch

### ISSUE THREAD

**Issue by azhangmn** [voxelmorph/voxelmorph#332]
```
def loss(self, _, y_pred):
        dy = torch.abs(y_pred[:, :, 1:, :, :] - y_pred[:, :, :-1, :, :])
        dx = torch.abs(y_pred[:, :, :, 1:, :] - y_pred[:, :, :, :-1, :])
        dz = torch.abs(y_pred[:, :, :, :, 1:] - y_pred[:, :, :, :, :-1])
```

When I input an array of shape `[1, 1, 1024, 1024]` as the input image, the output gradient image is shape `[1, 2, 1024, 1024]`, which makes sense to me. However, the loss function references 5 dimensions, instead of 4 dimensions. Is there a workaround to this, or is my input shape incorrect? Thanks!

**Comment by McHz1s** [comment]
Also occurred to me.
Make a little change that:
def func_name(...)
  pred_shape = y_pred.shape
  ...
  other codes
  ...
  dy_mean = 0 if pred_shape[2] == 1 or len(pred_shape) == 4 else torch.mean(dy)
  d = torch.mean(dx) + torch.mean(dz) + dy_mean
  ...
  return grad
